Replaced print statements with logging in worker/remote_transcoder.py

hamid26126 opened the pull request:
This PR addresses #378 by replacing all print() statements in worker/remote_transcoder.py with Python's logging module to improve log handling and maintainability.

Added logger = logging.getLogger(__name__) at the top of the file.
Replaced all print() calls with appropriate log levels (info, debug, warning, error).
Manually checked the file after making changes to ensure correctness.
No functional changes were made; only logging statements were updated.

This ensures logs can be filtered, formatted, and integrated with external log aggregation systems.

filthyrake reviewed (commented):
Hey @hamid26126! Thanks for tackling issue #378 - this is a really solid contribution! 🎉

The migration from `print()` to Python's `logging` module looks great overall. You've correctly replaced all 96 print statements and followed the existing codebase pattern with `logging.getLogger(__name__)`.

I have a few suggestions that would make this even better:

## A Couple of Things to Tweak

**1. Inconsistent log level for HLS validation failures**

There's a small inconsistency in how validation failures are logged:
- Line ~349 uses `logger.error()` for the original quality
- Line ~541 uses `logger.warning()` for other qualities

Since both are validation failures, it would be more consistent to use `logger.error()` for both.

**2. Retry messages logged as errors**

Around lines 793-794:
```python
logger.error(f"    Completion failed (attempt {attempt + 1}/{COMPLETE_JOB_MAX_RETRIES}): {error_msg}")
logger.error(f"    Retrying in {COMPLETE_JOB_RETRY_DELAY}s...")
```

For retryable failures, `logger.warning()` is typically more appropriate - it signals "something went wrong but we're handling it." Reserving `logger.error()` for the final failure (when all retries are exhausted) helps operators distinguish between transient issues and actual problems that need attention.

**3. Help message logged as warning**

Line ~1163:
```python
logger.warning("Register a worker first: curl -X POST...")
```

This is actually helpful information telling the user what to do, so `logger.info()` would be a better fit here.

## Nice-to-Have (Optional)

- The `import logging` could be moved up to the standard library imports section (around line 26) to follow Python import conventions
- You could remove the redundant "ERROR:" prefix from messages like `logger.error("ERROR: VLOG_WORKER_API_KEY...")` since the log level already indicates it's an error

---

Once the first three items are addressed, this will be good to merge! Thanks again for the contribution - having proper logging in the worker will make debugging and monitoring much easier. 🙏

hamid26126 commented:
@filthyrake 
Thanks for the review!
I’ve addressed all the suggested changes:

Made HLS validation failures consistent
Downgraded retry logs from error to warning
Changed the help message log level to info
Resolved the merge conflict and ensured logging is used instead of print

Please let me know if anything else needs adjustment.
